CafeTime/CafeTime/AuthManager.swift

```diff
@@ -25,85 +57,104 @@ class AuthManager {
             print("You have successfully logged in")
             
             if rememberUser {
-                GlobalSaver.shared.saveUser(email: email, password: password)
+                GlobalSaver.shared.saveLocalUserCredentials(email: email, password: password)
             }
             
             completion("", true)
         }
     }
-
-    func createUser(email: String, password: String, rememberUser: Bool, completion: @escaping (_ error: String, _ success: Bool) -> Void)  {
+    
+    func createUser(email: String, name: String, numberOfTables: String, country: String, foodType: String, password: String, rememberUser: Bool, completion: @escaping (_ error: String, _ success: Bool) -> Void)  {
         
-        FIRAuth.auth()?.createUser(withEmail: email, password: password, completion: { (user, error) in
+        FIRAuth.auth()?.createUser(withEmail: email, password: password, completion: { [weak self] (user, error) in
+            
+            guard let weakSelf = self else { return }
             
             if error != nil {
```

Comment:
you can use "if let " construction in this case
if let err = error{
       completion(err.localizedDescription, false)
}

in this case err will be not optional, and if it's nil you will skip this block